docs/http.md

```diff
@@ -1,79 +1,239 @@
 # HTTP API
 
-When a Cog Docker image is run, it serves an HTTP API for making predictions. For more information, take a look at [the documentation for deploying models](deploy.md).
+> [!TIP]
+> For information about how to run the HTTP server, 
+> see [our documentation to deploying models](deploy.md).
+
+When you run a Docker image built by Cog, 
+it serves an HTTP API for making predictions.
+
+The server supports both synchronous and asynchronous prediction creation:
+
+- **Synchronous**: 
+  The server waits until the prediction is completed
+  and responds with the result.
+- **Asynchronous**: 
+  The server immediately returns a response 
+  and processes the prediction in the background. 
+
+The client can create a prediction asynchronously
+by setting the `Prefer: respond-async` header in their request.
+When provided, the server responds immediately after starting the prediction 
+with `202 Accepted` status and a prediction object in status `processing`.
+
+> [!NOTE]
+> The only supported way to receive updates on the status of predictions
+> started asynchronously is using [webhooks](#webhooks). 
+> Polling for prediction status is not currently supported.
+
+The server also provides a way to create predictions idempotently,
```

Comment:
I want to make this a bit more user-active, but without losing any technical nuance. What do you think of this?

```suggestion
You can also use certain server endpoints to create predictions idempotently,
```